src/ThemeProvider.tsx

```diff
@@ -2,31 +2,51 @@ import PropTypes from 'prop-types';
 import * as React from 'react';
 import { useContext, useMemo } from 'react';
 
-export interface ThemeProviderProps {
-  prefixes: Record<string, unknown>;
+export interface ThemeContextValue {
+  prefixes: Record<string, string>;
+  dir?: string;
 }
 
-const ThemeContext = React.createContext({});
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ThemeProviderProps extends Partial<ThemeContextValue> {}
+
+const ThemeContext = React.createContext<ThemeContextValue>({ prefixes: {} });
 const { Consumer, Provider } = ThemeContext;
 
-function ThemeProvider({ prefixes, children }) {
-  const copiedPrefixes = useMemo(() => ({ ...prefixes }), [prefixes]);
+const ThemeProvider: React.FC<Partial<ThemeContextValue>> = ({
```

Comment:
what's the reason btw for switching to an arrow function here? I find the typing slightly harder to understand vs a plain function